method.rb

```diff
@@ -0,0 +1,218 @@
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator &&= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == false && parameter.nil?
+      loop do
+        break if i == size || status_cumulator == false
+
+        status_cumulator &&= !self[i].nil?
+        i += 1
+      end
+    elsif bg == false && parameter.is_a?(Regexp)
+      loop do
+        status_cumulator &&= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    elsif bg == false && parameter.is_a?(Class)
+      loop do
+        status_cumulator &&= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    else
+      loop do
+        status_cumulator &&= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == false
+      end
+    end
+    status_cumulator
+  end
+
+  def my_any?(parameter = nil) # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/MethodLength, Metrics/AbcSize
+    status_cumulator = false
+    i = 0
+    bg = block_given?
+    if bg == true && parameter.nil?
+      loop do
+        status_cumulator ||= yield self[i]
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter.is_a?(Regexp)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter.is_a?(Class)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == true && parameter_condition?(parameter)
+      puts 'warning: given block not used'
+      loop do
+        status_cumulator ||= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == false && parameter.nil?
+      loop do
+        break if i == size || status_cumulator == true
+
+        status_cumulator ||= !self[i].nil?
+        i += 1
+      end
+    elsif bg == false && parameter.is_a?(Regexp)
+      loop do
+        status_cumulator ||= !parameter.match(self[i]).nil?
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    elsif bg == false && parameter.is_a?(Class)
+      loop do
+        status_cumulator ||= self[i].is_a?(parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    else
+      loop do
+        status_cumulator ||= (self[i] == parameter)
+        i += 1
+        break if i == size || status_cumulator == true
+      end
+    end
+    status_cumulator
+  end
+
+  def my_none?(parameter = nil)
+    !my_any?(parameter)
```

Comment:
I strongly advise you against this. Not only you are not making yourself think about the solution to this, but you are also depending on another method. This is not a good practice and you should refactor your code.

- Right now, you method returns false when given a truthy block of code. It should return false.
- Your method doesn't consider the possibility when no block or argument is given. It should return true if the enumerable contains no truthy value.